File: publication-figure-renderer/r/tests/testthat/test-output_writer.R
test_that("writes the SVG text to the requested path as UTF-8", {
  svg_text <- "<svg xmlns=\"http://www.w3.org/2000/svg\"><text>Caf\u00e9</text></svg>"
  path <- tempfile(fileext = ".svg")
  on.exit(unlink(path), add = TRUE)

  returned <- write_svg(svg_text, path)

  expect_equal(returned, path)
  expect_true(file.exists(path))
  round_tripped <- readChar(path, file.info(path)$size, useBytes = TRUE)
  Encoding(round_tripped) <- "UTF-8"
  expect_equal(round_tripped, svg_text)
})

File: publication-figure-renderer/python/tests/test_output_writer.py
from pathlib import Path

from publication_figure_renderer.output_writer import write_svg


def test_writes_svg_text_to_the_requested_path_as_utf8(tmp_path: Path) -> None:
    svg_text = "<svg xmlns=\"http://www.w3.org/2000/svg\"><text>Café</text></svg>"
    target = tmp_path / "fig.svg"

    returned = write_svg(svg_text, target)

    assert returned == target
    assert target.exists()
    assert target.read_text(encoding="utf-8") == svg_text
